docs/concepts/primer.rst

```diff
@@ -258,14 +259,52 @@ Unsafe Code
 ~~~~~~~~~~~
 
 The CLR enables the ability to acccess native memory and do pointer
-arithmetic. These operations are needed for some algortithms and for
-calling some native APIs. The use of these capabilities is discouraged,
-since you no longer get the benefit of verifiability, nor will your code
-be allowed to run in all environments. The best practice is to confine
-unsafe code as much as possible and that the vast majority of code is
-type-safe.
+arithmetic via ``unsafe`` code. These operations are needed for certain algorithms and system interoperability.  Although powerful, use of unsafe code is discouraged unless it is necessary to interop with system APIs or implement the most efficient algorithm.  Unsafe code is not guaranteed to run in all environments, and also loses the benefits of a garbage collector and type safety.  It's recommended to confine
```

Comment:
Do you have an example of where unsafe code doesn't run? It makes it sounds like you could build an app this way and it might unpredictably not run. It's the "not guaranteed to run" part that is a bit confusing. Do you mean "no supported"?